### DISCUSSION TITLE
Filament table - Prevent user from select all for bulk action

### DISCUSSION THREAD

**Discussion by markvdlaan93** [filamentphp/filament#5608] [Ideas]
Good morning,

I have a table with more than 3 million entries. I've implemented bulk actions which works just fine, however the user has now the possibility to press on select all. I want to prevent this because it takes way too much time to select them all and the underlying functionality is not well suited for these kind of volumes.

I don't see any possibility to turn off the select all, am I correct? It would be nice if this is configurable through a closure in the bulk action so that it becomes available when the user reduced the amount of rows in the table by selecting filters.

**Comment by danharrin** [comment]
Feel free to PR an option to disable that button

**Reply by muesker** [reply]
PR is here:
https://github.com/filamentphp/filament/pull/17964
